tests/notebooks/training/training-integration.ipynb

```diff
@@ -1,918 +1,922 @@
+   "metadata": {},
+   "source": [
+    "### Initialise Training Client\n",
+    "\n",
+    "We will be using the Training SDK for any actions executed as part of this example."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "client = TrainingClient()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define Helper to print training logs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def print_training_logs(client, job_name: str, container: str, is_master: bool = True):\n",
+    "    logs = client.get_job_logs(name=job_name, container=container, is_master=is_master)\n",
+    "    print(logs)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define Helper to check that Job succeeded"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@retry(\n",
+    "    wait=wait_exponential(multiplier=2, min=1, max=30),\n",
+    "    stop=stop_after_attempt(50),\n",
+    "    reraise=True,\n",
+    ")\n",
+    "def assert_job_succeeded(client, job_name, job_kind):\n",
+    "    \"\"\"Wait for the Job to complete successfully.\"\"\"\n",
+    "    assert client.is_job_succeeded(\n",
+    "        name=job_name, job_kind=job_kind\n",
+    "    ), f\"Job {job_name} was not successful.\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define Helper to get the spec labels\n",
+    "This will add the label for the proxy PodDefault if the proxy envs are set"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "training_labels = {}\n",
+    "if os.environ.get('HTTP_PROXY') and os.environ.get('HTTPS_PROXY') and os.environ.get('NO_PROXY'):\n",
+    "    training_labels = {\"notebook-proxy\": \"true\"}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Test TFJob"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define a TFJob\n",
+    "\n",
+    "Define a TFJob object before deploying it."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "TFJOB_NAME = \"mnist7\"\n",
+    "TFJOB_CONTAINER = \"tensorflow\"\n",
+    "TFJOB_IMAGE = \"kubeflow/tf-mnist-with-summaries:latest\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "container = V1Container(\n",
+    "    name=TFJOB_CONTAINER,\n",
+    "    image=TFJOB_IMAGE,\n",
+    "    command=[\n",
+    "        \"sh\", \"-c\",\n",
+    "        # Download MNIST dataset using curl to overcome proxy problems\n",
```

Comment:
can we link here issue https://github.com/canonical/training-operator/issues/182